src/credentials/tests/BUILD.gn

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("//build_overrides/nlunit_test.gni")
+
+import("${chip_root}/build/chip/chip_test_suite.gni")
+
+chip_test_suite("tests") {
+  output_name = "libChipCredentials"
+
+  sources = [
+    "CHIPCert_test_vectors.cpp",
+    "CHIPCert_test_vectors.h",
+    "TestChipCert.cpp",
```

Comment:
you can use `test_sources` and not write your own driver. That has the benefit of easier integration into QEMU and ZEPHYR unit tests